### DISCUSSION TITLE
request bulk ignored type casting for compute result  to small types(byte,sbyte,short,ushort)

### DISCUSSION THREAD

**Discussion by sgf** [dotnet/csharplang#2596] [Language Ideas]
here is some examples.
before
```C#
       sbyte b1 = 200;
       byte b2 = (byte)(b1 + 60);
       short s3 = (short)(b1 - 60);
       ushort i3 = (ushort)(b1 * 60);
       sbyte l5 = (sbyte)(b1 / 60); 
       int intRlt=(int)(99999L+9999L); 
```
developers know what they're doing.The unchecked{} is not work.
Suggestion: let unchecked work for this (follow code)?
```C#
unchecked
{
       sbyte b1 = 200;
       byte b2 = b1 + 60;
       short s3 = b1 - 60;
       ushort i3 = b1 * 60;
       byte l5 = b1 / 60; 
       int intRlt=99999L+9999L; 
}
```
Need a way to Bulk ignored type casting. to avoid write the follow code every where.
```C#
smallNum= (smallNum)(smallNum +-*/ smallNum)
 //or (smallNum +-*/ smallNum)  or (smallNum +-*/  LargeNum)
```

**Comment by canton7** [comment]
> developers know what they're doing.

I'd argue that if a developer writes `sbyte b1 = 200;`, it's probably a mistake. In this case, at least, I think the explicit unchecked cast should always be necessary.

For the others, the problem is not with checked/unchecked, but the fact that `short + short = int`, and there's no implicit conversion from `int` to `short`. While I can sympathise with the issue here, I don't think that `unchecked` is the right syntax to try and use.

**Comment by sgf** [comment]
> > developers know what they're doing.
> 
> I'd argue that if a developer writes `sbyte b1 = 200;`, it's probably a mistake. In this case, at least, I think the explicit unchecked cast should always be necessary.
> 
> For the others, the problem is not with checked/unchecked, but the fact that `short + short = int`, and there's no implicit conversion from `int` to `short`. While I can sympathise with the issue here, I don't think that `unchecked` is the right syntax to try and use.

😂 I'm sorry, this example  `  sbyte b1 = 200;` is not very rigorous. 
my means like ur says: `short + short` 
smallNum= (smallNum)(smallNum +-*/ smallNum)
 //or (smallNum +-*/ smallNum)  or (smallNum +-*/  LargeNum)

also << >> | $
